blockstore/src/main/java/brooklyn/location/blockstore/azure/arm/AzureArmVolumeManager.java

```diff
@@ -0,0 +1,411 @@
+        return vmApi.get(vmName) != null;
+    }
+
+    private boolean waitForJobToComplete(final AzureComputeApi api, final URI uri, Duration waitTime) {
+        checkNotNull(uri, "uri must not be null");
+        final JobApi jobApi = api.getJobApi();
+        final AtomicReference<JobStatus> latest = new AtomicReference<JobStatus>();
+        
+        boolean done = Repeater.create("Waiting job for URI to completee: " + uri)
+                .every(1, TimeUnit.SECONDS)
+                .limitTimeTo(waitTime)
+                .until(new Callable<Boolean>() {
+                    @Override
+                    public Boolean call() throws Exception {
+                        JobStatus currentJobStatus = jobApi.jobStatus(uri);
+                        latest.set(currentJobStatus);
+                        return currentJobStatus == ParseJobStatus.JobStatus.DONE;
+                    }
+                })
+                .run();
+        return done;
+    }
+
+    private static class AzureArmBlockDevice implements BlockDevice {
+
+        private static final Logger LOG = LoggerFactory.getLogger(AzureArmBlockDevice.class);
+
+        private final JcloudsLocation location;
+        private final DataDisk disk;
+        private final String resourceGroupName;
+        private final String storageAccountName;
+        
+        private AzureArmBlockDevice(JcloudsLocation location, DataDisk disk, String resourceGroupName, String storageAccountName) {
+            this.location = checkNotNull(location, "location");
+            this.disk = checkNotNull(disk, "disk");
+            this.resourceGroupName = checkNotNull(resourceGroupName, "resourceGroupName");
+            this.storageAccountName = checkNotNull(storageAccountName, "storageAccountName");
+        }
+
+        @Override
+        public String getId() {
+            return disk.name();
+        }
+
+        @Override
+        public JcloudsLocation getLocation() {
+            return location;
+        }
+
+        public DataDisk getDisk() {
+            return disk;
+        }
+
+        public String getResourceGroupName() {
+            return resourceGroupName;
+        }
+        
+        public String getStorageAccountName() {
+            return storageAccountName;
+        }
+        
+        @Override
+        public AzureArmAttachedBlockDevice attachedTo(JcloudsMachineLocation machine, String deviceName) {
+            if (!machine.getParent().equals(location)) {
+                LOG.warn("Attaching device to machine in different location to its creation: id={}, location={}, machine={}",
+                        new Object[]{getId(), location, machine});
+            }
+            return new AzureArmAttachedBlockDevice(machine, disk, resourceGroupName, storageAccountName, deviceName);
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
```

Comment:
On balance, I think it's fine that we use this - the advantage is we (concisely) get a consistent toString representation, without inventing our own format or duplicating code that is in guava.

As for "when not strictly needed", it's tricky to know where the line is. At one extreme, nothing in guava is "needed" because it's "just code" (which I'm sure we'll all agree is a ridiculous argument). In the middle somewhere, it makes our code easier to write, read and maintain because it's using consistent concise well-documented patterns that other very smart people are maintaining. At the other extreme, it's exposing us to problems where guava changes its methods etc, when we could have achieved the same thing in the same number of lines of code.

To me, the benefits of using guava like this almost always outweigh the cost when the guava team change something.

But saying that, the one place we have over-used guava is when we expose it in the public API of brooklyn (as is done in jclouds). Unfortunately, that would be very hard to undo, and would almost certainly cause serious backwards compatibility problems for java-users.